### ISSUE TITLE
Show example that uses DbFactory

### ISSUE THREAD

**Issue by KurtP20** [dotnet/AspNetCore.Docs#26873]
[Update by guardrex to change the metadata.]

Could you please add an example where Identity uses a `DbFactory` instead of the `DbContext`? Using a factory is the recommended way in EF Core >5.0, see [link1](https://docs.microsoft.com/en-us/ef/core/what-is-new/ef-core-5.0/whatsnew#dbcontextfactory), [link2](https://docs.microsoft.com/en-us/ef/core/dbcontext-configuration/#using-a-dbcontext-factory-eg-for-blazor). The example on this page (and the template for new Blazor server side projects) uses
```
Services.AddEntityFrameworkStores<ApplicationDbContext>()
```
to pass the `DbContext` to Identity. There is a simple solution described [here](https://stackoverflow.com/questions/64169025/using-identity-with-adddbcontextfactory-in-blazor), which passes the `DbContext` obtained from the `DbFactory`:
```
services.AddScoped<ApplicationDbContext>(p => p.GetRequiredService<ApplicationDbContextFactory<ApplicationDbContext>>().CreateDbContext());
```

**But**, as described in detail [here](https://stackoverflow.com/questions/73498604/dependency-injected-usermanager-causes-concurrency-issue-how-to-make-usermanage), the `UserManager` obtained via dependency injection is the same in every Blazor component, which can lead to a concurrency issue because asynchronous components typically run on differnt threads.

I think that Identity should also accept a `DbContextFactory` to enable something like `services.AddScoped<ApplicationDbContextFactory>()` . So it might need more updates than just the documentaion...

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 4a101e4d-85a2-8013-0f62-40e13e45b3c6
* Version Independent ID: 66a0a920-ebd8-9b1d-c8c5-953ebb766de0
* Content: [ASP.NET Core Blazor Server with Entity Framework Core (EF Core)](https://docs.microsoft.com/en-us/aspnet/core/blazor/blazor-server-ef-core?view=aspnetcore-6.0)
* Content Source: [aspnetcore/blazor/blazor-server-ef-core.md](https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/blazor/blazor-server-ef-core.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-blazor**
* GitHub Login: @guardrex
* Microsoft Alias: **jeliknes**

**Comment by guardrex** [comment]
Hello @KurtP20 ... I'll take this. We're working most of the Blazor content into the Blazor docs node, so anything having to do with EFCore and Blazor will be taken care of there, especially since services in Blazor apps are bit different than in normal ASP.NET Core apps.

**Comment by guardrex** [comment]
Quick reaction ... gotta get some 🍗 ... I just walked in the door from the 🏋️‍♂️ gym where I ran a respectable 800 calories off on the treadmill ... trying to drop the *Pandemic Poundage!&trade;* 🤣 

Did you see the *New DbContext instances* section at ...

https://docs.microsoft.com/aspnet/core/blazor/blazor-server-ef-core#new-dbcontext-instances

... in the Blazor-EFCore doc?

Furthermore, did you also note the guidance in the last paragraph of the opening remarks at ...

https://docs.microsoft.com/en-us/aspnet/core/blazor/security/

... and the idea is that you can use `UserManager`/`SignInManager` server-side (controller, for example). True tho, not in a Razor component. You're correct about that.

**Comment by guardrex** [comment]
btw -- Even if the content there (perhaps along with the sample app for that topic) cover it, **_leave this issue open_**. I see a problem in that section. It doesn't show/explain the actual injection of the factory piece. I'd like to show ...

```razor
@inject IDbContextFactory<ContactContext> DbFactory
```

... and the text does say what `Wrapper` is and cross-links to the sample app. However, it would be easier/faster to consume the example in the section (i.e., `Pages/Index.razor`) if I added more API from the component and try to avoid readers having to go look at the sample app. I'm going to mark this issue for a little work.

**Comment by KurtP20** [comment]
Hi Luke, so you are saying that office is better than home office in terms of 🏋️ 😉 

Your first link suggests to use `new DbContext` but it also recommends to use a factory instead. Since I inject the `UserManager`, I would not know how to tell `UserManager` to use the `DbContext` I created with `new` or via factory. 

I was also aware of the last sentence in your second link

> ASP.NET Core abstractions, such as [SignInManager<TUser>](https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.identity.signinmanager-1) and [UserManager<TUser>](https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.identity.usermanager-1), aren't supported in Razor components.

but honestly I don't understand it, because `UserManager` does work in a Razor component, I am using it. The only drawback I encountered so far is the concurrency issue mentioned in this issue. So in some sense I think this paragraph is wrong, and because it probably isn't, I just don't understand it properly. Maybe I also mix up the terms, I find it very confusing with Razor pages, Razor components that are called Blazor components but use Razor syntax and Razor pages that actually are Razor components 😵 

Mentioning that one can also inject a `DbFactory` would be good. In my app I only inject services that encapsulate all DB access, so I am not affectd by that.

Thanks for taking on that issue and a motivated 🏃 over lunch.
